src/test/common.node.ts

```diff
@@ -320,3 +333,39 @@ export async function captureScreenShot(fileNamePrefix: string) {
         console.error(`Failed to capture screenshot into ${filename}`, ex);
     }
 }
+
+export function initializeCommonNodeApi() {
+    initializeCommonApi({
+        async createTemporaryFile(options: {
+            contents?: string;
+            extension: string;
+        }): Promise<{ file: Uri } & IDisposable> {
+            const extension = options.extension || '.py';
+            const tempFile = tmp.tmpNameSync({ postfix: extension });
+            if (options.contents) {
+                await fs.writeFile(tempFile, options.contents);
+            }
+            return { file: Uri.file(tempFile), dispose: () => swallowExceptions(() => fs.unlinkSync(tempFile)) };
+        },
+        async startJupyterServer(notebook?: NotebookDocument, useCert: boolean = false): Promise<any> {
```

Comment:
We end up duplicating the test files because some functions like StartJupyter, createTempFile are specific to web vs desktop.
Created a common layer so that we call the common function and inject the actual implementation during runtime.
This way we don't need to duplicate files or the like. I.e. we can write some code that will run in all places (web & desktop tests)